# 1. introduction

## 1.1 about dataset

<p>this dataset is very suitable for analyzing the fuel consumption of cars due to a number of features. We will look at these features and analyze them.</p>

# 2. understand dataset

## 2.1 importing needed libraries

In [2]:
import pandas as pd
import matplotlib.pyplot as plt

## 2.2 reading dataset file 

In [10]:
df = pd.read_csv("./mpg.csv")

## 2.3 get a summary about dataset varables 

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 234 entries, 0 to 233
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   manufacturer  234 non-null    object 
 1   model         234 non-null    object 
 2   displ         234 non-null    float64
 3   year          234 non-null    int64  
 4   cyl           234 non-null    int64  
 5   trans         234 non-null    object 
 6   drv           234 non-null    object 
 7   cty           234 non-null    int64  
 8   hwy           234 non-null    int64  
 9   fl            234 non-null    object 
 10  class         234 non-null    object 
dtypes: float64(1), int64(4), object(6)
memory usage: 20.2+ KB


## 2.4 gain some data about dataset

In [13]:
df.head()

manufacturer model  displ  year  cyl       trans drv  cty  hwy fl    class
0         audi    a4    1.8  1999    4    auto(l5)   f   18   29  p  compact
1         audi    a4    1.8  1999    4  manual(m5)   f   21   29  p  compact
2         audi    a4    2.0  2008    4  manual(m6)   f   20   31  p  compact
3         audi    a4    2.0  2008    4    auto(av)   f   21   30  p  compact
4         audi    a4    2.8  1999    6    auto(l5)   f   16   26  p  compact

In [14]:
df.shape

(234, 11)

In [8]:
df.describe().T

count         mean       std     min     25%     50%     75%     max
displ  234.0     3.471795  1.291959     1.6     2.4     3.3     4.6     7.0
year   234.0  2003.500000  4.509646  1999.0  1999.0  2003.5  2008.0  2008.0
cyl    234.0     5.888889  1.611534     4.0     4.0     6.0     8.0     8.0
cty    234.0    16.858974  4.255946     9.0    14.0    17.0    19.0    35.0
hwy    234.0    23.440171  5.954643    12.0    18.0    24.0    27.0    44.0

In [15]:
df.columns

Index(['manufacturer', 'model', 'displ', 'year', 'cyl', 'trans', 'drv', 'cty',
       'hwy', 'fl', 'class'],
      dtype='object')

## 2.4.1 analysis result

1. **dataset Overview**:
    - dataset contains 11 columns and 234 rows
    - dataset contains both numerical and categorical variables

2. **keys**:
    - *manufacturer*: the producer company name
    - *model*: the car model 
    - *displ*: car engine capacity (displacement in liters)
    - *year*: year of manufacture of the car
    - *cyl*: number of car cylinders
    - *trans*: is the car auto or manual? (transmission type)
    - *drv*: drive train type (f = front-wheel drive, r = rear-wheel drive, 4 = 4-wheel drive)
    - *cty*: distance the car travels in the city with one gallon of gasoline (miles per gallon)
    - *hwy*: distance the car travels on the highway with one gallon of gasoline (miles per gallon)
    - *fl*: car fuel type (e = ethanol, d = diesel, r = regular, p = premium, c = CNG)
    - *class*: car class/type (compact, midsize, SUV, 2seater, minivan, pickup, subcompact)


In [19]:
pd.unique(df['cyl'])

array([4, 6, 8, 5])